testing/unit-testing-jest/docs/http-test.md

```diff
@@ -0,0 +1,105 @@
+
+## HttpClientTestingModule
+To test services with the [HttpClient](https://angular.io/api/common/http/HttpClient) dependency, we can use the utilities of Angular testing. We don't need to use all the object utilities of HttpClient, so we use [HttpClientTestingModule](https://angular.io/api/common/http/testing/HttpClientTestingModule#description) to test, this is a soft module designed just for testing. For example:
+```ts
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Car } from '../models/cars';
+
+@Injectable({ providedIn: 'root' })
+export class CarService {
+  constructor(private http: HttpClient) {}
+
+  findCarById(carId: number): Observable<Car> {
+    return this.http.get<Car>(`api/cars/${carId}`);
+  }
+
+  findAllCars(): Observable<Car[]> {
+    return this.http.get('api/cars') as Observable<Car[]>;
+  }
+
+  saveCar(carId: number, changes: Partial<Car>): Observable<Car> {
+    return this.http.put<Car>(`api/cars/${carId}`, changes);
+  }
+}
+```
+
+```ts
+import { TestBed } from '@angular/core/testing';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { CARS } from '../data/data';
+import { CarService } from './car.service';
+import { of } from 'rxjs';
+
+describe('CarService Using HttpClientTestingModule', () => {
+  let service: CarService;
+  let controller: HttpTestingController;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [],
+    });
+
+    service = TestBed.inject(CarService);
```

Comment:
What is `.inject`?

Reply:
`.inject` is the lates feature to get the instance of a dependency. But, this doesn't means that we are injecting the dependency again.

Reply:
Great, I think we could add it to the guide 💯